wordpress-mobile/WordPress-iOS
# [Stats Refresh] Finish insights core data persistence.

heyo, another layer done. (goes hand in hand with https://github.com/wordpress-mobile/WordPressKit-iOS/pull/125)

This "teaches" `StatsInsightStore` about how to read/write the Core Data objects related to Stats Insights — I expected the finished design to be slightly cleaned-up and generalised out to the `Store` API — but I'm not sure how that'll look with the time-series data, so I didn't want to pre-maturely freeze an API here. 

Also included here (and bulk of this PR by volume) is mapping from Core Data values back into immutable `struct`s that WPKit provides — this was done so that the layers above `StatsStore` don't have to have any awareness of the fact that the data is persisted using Core Data — as far as the rest of the app is concerned, Stats is just a bunch of `struct`s flying around.

To test:
This is actually quite tricky atm, because of how the stats are set up. For testing purposes, I suggest going to `StatsViewController.m` and temporarily commenting out lines 115-120. Then:

1. First, go to Stats page on any site (hogwarts is good, since it's fairly high traffic)
2. Let all data download
3. Background the app
4. Bring it into foreground
5. Close the app
6. Turn off WI-Fi
7. Go to Stats for that site again
8. Verify that there's data displayed ✨ 


@aerych with other Stats folks AFK and/or on Support rotation, would you have a second to take a peek at it?
